src/invidious/helpers/helpers.cr

```diff
@@ -68,7 +68,7 @@ class Config
   property feed_threads : Int32 = 1                # Number of threads to use for updating feeds
   property output : String = "STDOUT"              # Log file path or STDOUT
   property log_level : LogLevel = LogLevel::Info   # Default log level, valid YAML values are ints and strings, see src/invidious/helpers/logger.cr
-  property db : DBConfig                           # Database configuration
+  property db_url : String = "postgres://kemal:kemal@localhost:5432/invidious" # Database configuration
```

Comment:
There's probably a way to convert this to an URI directly here instead of calling `URI.parse` in a separate step?

Reply:
You could subclass `URI` and make it YAML serializable.

- https://crystal-lang.org/api/0.35.1/Object.html#from_yaml(string_or_io:String%7CIO)-class-method
- Enum example:
  - https://github.com/crystal-lang/crystal/blob/5999ae29b/src/yaml/from_yaml.cr#L222
  - https://github.com/crystal-lang/crystal/blob/5999ae29b/src/yaml/to_yaml.cr#L109

Could also be added to `URI` itself.

Reply:
Better example:
- https://github.com/iv-org/invidious/blob/ba148d749f3f3b745704c35bde2f72ea194b1ddb/src/invidious/helpers/helpers.cr#L93
- https://github.com/iv-org/invidious/blob/ba148d749f3f3b745704c35bde2f72ea194b1ddb/src/invidious/users.cr#L146